src/delta.c

```diff
@@ -159,22 +159,31 @@ static rs_result rs_delta_s_scan(rs_job_t *job)
             result=rs_appendmatch(job,match_pos,match_len);
             RollsumInit(&job->weak_sum);
         } else {
-            /* rotate the weak_sum and append the miss byte */
-            RollsumRotate(&job->weak_sum,job->scoop_next[job->scoop_pos],
+            if (!job->block_match) {
+                /* rotate the weak_sum and append the miss byte */
+                RollsumRotate(&job->weak_sum,job->scoop_next[job->scoop_pos],
                           job->scoop_next[job->scoop_pos+job->block_len]);
-            result=rs_appendmiss(job,1);
-            if (rs_roll_paranoia) {
-                RollsumInit(&test);
-                RollsumUpdate(&test, job->scoop_next+job->scoop_pos,
+                result=rs_appendmiss(job,1);
+                if (rs_roll_paranoia) {
+                    RollsumInit(&test);
+                    RollsumUpdate(&test, job->scoop_next+job->scoop_pos,
                               job->block_len);
-                if (RollsumDigest(&test) != RollsumDigest(&job->weak_sum)) {
-                    rs_fatal("mismatch between rolled sum %#x and check %#x",
+                    if (RollsumDigest(&test) != RollsumDigest(&job->weak_sum)) {
+                        rs_fatal("mismatch between rolled sum %#x and check %#x",
                              (int)RollsumDigest(&job->weak_sum),
                              (int)RollsumDigest(&test));
+                    }
                 }
-                
+            } else {
+                /* append block_len miss bytes */
+                result=rs_appendmiss(job,job->block_len);
+                RollsumInit(&job->weak_sum);
```

Comment:
Where is the weak_sum actually computed? Only RollsumInit seems to get called here.